Sources/MapboxCoreNavigation/BillingHandler.swift

```diff
@@ -0,0 +1,400 @@
+
+    func pauseBillingSession(for sessionType: BillingHandler.SessionType) {
+        let skuToken = tripSku(for: sessionType)
+        BillingServiceNative.pauseBillingSession(for: skuToken)
+    }
+
+    func resumeBillingSession(for sessionType: BillingHandler.SessionType,
+                              onError: @escaping (BillingServiceError) -> Void) {
+        let skuToken = tripSku(for: sessionType)
+        BillingServiceNative.resumeBillingSession(for: skuToken) { nativeBillingServiceError in
+            onError(BillingServiceError(nativeBillingServiceError))
+        }
+    }
+
+    func stopBillingSession(for sessionType: BillingHandler.SessionType) {
+        let skuToken = tripSku(for: sessionType)
+        BillingServiceNative.stopBillingSession(for: skuToken)
+    }
+
+    func triggerBillingEvent(onError: @escaping (BillingServiceError) -> Void) {
+        BillingServiceNative.triggerBillingEvent(forAccessToken: accessToken,
+                                                 userAgent: userAgent,
+                                                 skuIdentifier: mauSku) { nativeBillingServiceError in
+            onError(BillingServiceError(nativeBillingServiceError))
+        }
+    }
+
+    func getSessionStatus(for sessionType: BillingHandler.SessionType) -> BillingHandler.SessionState {
+        switch BillingServiceNative.getSessionStatus(for: tripSku(for: sessionType)) {
+        case .noSession: return .stopped
+        case .sessionActive: return .running
+        case .sessionPaused: return .paused
+        @unknown default:
+            preconditionFailure("Unsupported session status from BillingServiceNative")
+        }
+    }
+
+    private func tripSku(for sessionType: BillingHandler.SessionType) -> SKUIdentifier {
+        switch sessionType {
+        case .activeGuidance:
+            return .nav2SesTrip
+        case .freeDrive:
+            return .nav2SesFDTrip
+        }
+    }
+}
+
+/**
+ Receives events about navigation changes and triggers appropriate events in `BillingService`.
+
+ Session can be paused (`BillingHandler.pauseBillingSession(with:)`),
+ stopped (`BillingHandler.stopBillingSession(with:)`) or
+ resumed (`BillingHandler.resumeBillingSession(with:)`).
+
+ State of the billing sessions can be obtained using `BillingHandler.sessionState(uuid:)`.
+ */
+final class BillingHandler {
+    /// Parameters on an active session.
+    private struct Session {
+        let type: SessionType
+        /// Indicates whether the session is active but paused.
+        var isPaused: Bool
+    }
+
+    /// The state of the billing session.
+    enum SessionState: Equatable {
+        /// Indicates that there is no active billing session.
+        case stopped
+        /// There is an active paused billing session.
+        case paused
+        /// There is an active running billing session.
+        case running
+    }
+
+    /// Supported session types.
+    enum SessionType: Equatable {
+        case freeDrive
```

Comment:
Maybe it's better to use `passiveNavigation` here? As we are trying not to use the `freeDrive` term in the code.
Sorry if it was already mentioned here.

Reply:
There are a lot of discussions about this term but there is no agreed name as of now. This is internal API and can be changed without any problems. I would like to stick with the term used in public docs for now.